infer/resource.go

```diff
@@ -547,11 +547,17 @@ func (rc *derivedResourceController[R, I, O]) Check(ctx p.Context, req p.CheckRe
 	// The user has not implemented check, so do the smart thing by default; We just check
 	// that we can de-serialize correctly
 	var i I
-	_, err := decode(req.News, &i, req.News.ContainsUnknowns())
+	secrets, err := decode(req.News, &i, req.News.ContainsUnknowns())
 	if err == nil {
+		if err := applyDefaults(&i); err != nil {
+			return p.CheckResponse{}, fmt.Errorf("unable to apply defaults: %w", err)
+		}
+
+		inputs, err := encode(i, secrets, false)
```

Comment:
This only edits Check so I'm guessing defaults for Invokes and for the provider itself (CheckConfig) are not covered by the PR?

Reply:
I was planning to follow up for those, but it would be better to make that explicit.
https://github.com/pulumi/pulumi-go-provider/issues/115
https://github.com/pulumi/pulumi-go-provider/issues/116